Add code snippets for image-text-to-text

Wauplin opened the pull request:
This PR adds code snippets for `image-text-to-text` models, say [meta-llama/Llama-3.2-11B-Vision-Instruct](https://huggingface.co/meta-llama/Llama-3.2-11B-Vision-Instruct?inference_api=true) for example :smile: 

I've tested all three examples locally and they work as expected :)

julien-c commented on packages/tasks/src/snippets/js.ts:
```diff
@@ -41,6 +41,35 @@ for await (const chunk of inference.chatCompletionStream({
+	],
+	max_tokens: 500,
+})) {
+	process.stdout.write(chunk.choices[0]?.delta?.content || "");
```
```suggestion
	process.stdout.write(chunk.choices[0]?.delta?.content);
```

julien-c commented on packages/tasks/src/snippets/curl.ts:
```diff
@@ -27,6 +27,32 @@ export const snippetTextGeneration = (model: ModelDataMinimal, accessToken: stri
 };
 
+export const snippetImageTextToTextGeneration = (model: ModelDataMinimal, accessToken: string): string => {
+	if (model.config?.tokenizer_config?.chat_template) {
```
use model.tags.include("conversational") to be consistent with what @mishig25 did? (i think)

mishig25 replied:
`model.tags.include("conversational")`

yes, thats what I did

Wauplin replied:
I took the same as what existed for `textGeneration`. If I try to access "tags", I'm getting

```
Property 'tags' does not exist on type 'ModelDataMinimal'.
```

See https://github.com/huggingface/huggingface.js/blob/fae585340d11fb4f7f034387e19e275134e8d882/packages/tasks/src/snippets/types.ts#L8

coyotte508 replied:
sure

Wauplin replied:
right, correct. Addressed in https://github.com/huggingface/huggingface.js/pull/927/commits/9cc3d868ec3fca027ec94cce3026f37bfc8c14ba

mishig25 commented on packages/tasks/src/snippets/python.ts:
```diff
@@ -17,6 +17,30 @@ for message in client.chat_completion(
+	"${model.id}",
+	token="${accessToken || "{API_TOKEN}"}",
+)
+image_url = "https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg"
```
maybe good to show an example with `pillo.Image.open` and use that image's base64 so that users can get an example where they can load local images

julien-c replied:
hmm i'd say this would complicate a bit too much (no strong opinion though)

Note that this is for remote inference not local usage

Wauplin replied:
do you know if it's possible to have several snippets by returning a list, same as for code snippets ?

mishig25 replied:
> do you know if it's possible to have several snippets by returning a list, same as for code snippets ?

for inference snippet, not possible right now. So suggest that:
1. we merge this PR as it is with only image url example
2. mayeb unify on moon-side to have inference snippet to be able to have a list like code snippets. If so, we can re-iterate and add an example with a local image

Wauplin commented:
@coyotte508 @mishig25 thanks for the feedback. I addressed above the comment to use `"conversational"` tag. Otherwise let's merge this and come back to it when multiple inference snippets can be provided. Can I have a final review before merging?

julien-c commented on packages/tasks/src/snippets/types.ts:
👍

mishig25 commented on packages/tasks/src/snippets/js.ts:
```diff
@@ -41,6 +41,35 @@ for await (const chunk of inference.chatCompletionStream({
+		return `import { HfInference } from "@huggingface/inference";
+
+const inference = new HfInference("${accessToken || `{API_TOKEN}`}");
+const image_url = "https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg"
```
nit: but in js, we usually prefer camelCase `imagUrl` rather than snake_case `image_url`. No?

Wauplin replied:
addressed in s/f6d7ff8a2be24abe60ded03ee4e7ec4eb2ed33f8

mishig25 commented on packages/tasks/src/snippets/js.ts:
```diff
@@ -41,6 +41,35 @@ for await (const chunk of inference.chatCompletionStream({
+		return `import { HfInference } from "@huggingface/inference";
+
+const inference = new HfInference("${accessToken || `{API_TOKEN}`}");
+const image_url = "https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg"
```
```suggestion
const image_url = "https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg";
```

nit: ending `;` is missing

Wauplin replied:
addressed in https://github.com/huggingface/huggingface.js/pull/927/commits/d9d5a4353d5ccf8c4fe47fe92217a4e869cecb48

mishig25 commented on packages/tasks/src/snippets/python.ts:
```diff
@@ -17,6 +17,30 @@ for message in client.chat_completion(
+export const snippetConversationalWithImage = (model: ModelDataMinimal, accessToken: string): string =>
+	`from huggingface_hub import InferenceClient
+
+client = InferenceClient(
```
this comment maybe applies to text conv snippet as well

mishig25 replied:
if we decide to change it, lets handle in subseq PR

Wauplin replied:
I'll use the same convention then. I've addressed it in https://github.com/huggingface/huggingface.js/pull/927/commits/e4c6cba687930534def4819f54484700781f8644 for both the text-generation and text-image-to-text snippets.

mishig25 commented:
github is showing that `model` is indendted wrongly?

<img width="381" alt="image" src="https://github.com/user-attachments/assets/af6e3495-16e6-4732-9f33-3c6d08bed03d">

Wauplin commented:
> github is showing that model is indendted wrongly?

Well well well, looks like it yes. Addressed in https://github.com/huggingface/huggingface.js/pull/927/commits/0f8452c3f8199cb3d0e4c9fd6a21b81c0a86e39b

mishig25 reviewed (approved):
lgtm! again

Wauplin commented:
Thanks! Sorry about the back and forth :grimacing:
